src/main/java/accesingdatapostgres/api/CarApi.java

```diff
@@ -0,0 +1,47 @@
+package accesingdatapostgres.api;
+
+import accesingdatapostgres.dao.entity.CarEntity;
+import accesingdatapostgres.service.CarService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping(path = "/cars")
+public class CarApi {
+    private final CarService carService;
+
+    @Autowired
+    public CarApi(CarService carService) {
+        this.carService = carService;
+    }
+
+    @PostMapping(path = "/add")
+    public CarEntity addCar(@RequestBody CarEntity carEntity) {
+
+        return this.carService.save(carEntity);
+    }
+
+    @PutMapping(path = "/update")
+    public CarEntity updateCar(@RequestBody CarEntity carEntity) {
+        return this.carService.save(carEntity);
+    }
+
+    @GetMapping(path = "/one/{id}")
+    public Optional<CarEntity> getOne(@PathVariable Integer id) {
```

Comment:
Optionalsów nie zwracamy z kontrollerów. W springu jest taki obiekt który pozwala Ci wszystkie parametry http zwrócić, wyglądałoby to tak ResponseEntity<CarDto>, zapoznaj się jak to budować